subsys/fb/cfb.c

```diff
@@ -235,35 +591,57 @@ static void draw_line(struct char_framebuffer *fb, int16_t x0, int16_t y0, int16
 }
 
 static int draw_text(const struct device *dev, const char *const str, int16_t x, int16_t y,
-		     bool wrap)
+		     bool print_cmd)
 {
 	const struct char_framebuffer *fb = &char_fb;
-	const struct cfb_font *fptr;
-
-	if (!fb->fonts || !fb->buf) {
-		return -ENODEV;
+	const struct cfb_font *fptr = &(fb->fonts[fb->font_idx]);
+	const uint16_t columns = (fb->x_res + fb->kerning) / (fptr->width + fb->kerning);
+	const uint16_t firstrow_columns =
+		((fb->x_res + fb->kerning) - x) / (fptr->width + fb->kerning);
+	const uint16_t draw_height =
+		(((strlen(str) - firstrow_columns) + columns) / columns + 1) * fptr->height;
+	const bool need_reverse = (((fb->screen_info & SCREEN_INFO_MONO_MSB_FIRST) != 0) !=
+				   ((fptr->caps & CFB_FONT_MSB_FIRST) != 0));
```

Comment:
TBH, it does not look good. Better to introduce few helpers, like
```
const struct cfb_font *fptr = cfb_get_font(fb, font_idx);
const uint16_t columns = cfb_get_columns(fb, 0);
const uint16_t firsrow_columns = cfb_get_columns(fb, x);
```
```c
....
		.need_reverse = cfb_needs_reverse(fb),
....
```

Reply:
Oh, you're right.
I made inline helper function and removed intermediate values.